openedx/core/djangoapps/xblock/runtime/learning_core_runtime.py

```diff
@@ -0,0 +1,299 @@
+        parsed OLX.
+        """
+        # We can do this more efficiently in a single query later, but for now
+        # just get it the easy way.
+        component = self._get_component_from_usage_key(usage_key)
+        component_version = component.versioning.draft
+        if component_version is None:
+            raise NoSuchUsage(usage_key)
+
+        content = component_version.contents.get(
+            componentversioncontent__key="block.xml"
+        )
+        xml_node = etree.fromstring(content.text)
+        block_type = usage_key.block_type
+        keys = ScopeIds(self.user_id, block_type, None, usage_key)
+
+        if xml_node.get("url_name", None):
+            log.warning("XBlock at %s should not specify an old-style url_name attribute.", usage_key)
+
+        block_class = self.mixologist.mix(self.load_block_type(block_type))
+
+        if hasattr(block_class, 'parse_xml_new_runtime'):
+            # This is a (former) XModule with messy XML parsing code; let its parse_xml() method continue to work
+            # as it currently does in the old runtime, but let this parse_xml_new_runtime() method parse the XML in
+            # a simpler way that's free of tech debt, if defined.
+            # In particular, XmlMixin doesn't play well with this new runtime, so this is mostly about
+            # bypassing that mixin's code.
+            # When a former XModule no longer needs to support the old runtime, its parse_xml_new_runtime method
+            # should be removed and its parse_xml() method should be simplified to just call the super().parse_xml()
+            # plus some minor additional lines of code as needed.
+            block = block_class.parse_xml_new_runtime(xml_node, runtime=self, keys=keys)
+        else:
+            block = block_class.parse_xml(xml_node, runtime=self, keys=keys, id_generator=None)
+
+        # Update field data with parsed values. We can't call .save() because it will call save_block(), below.
+        block.force_save_fields(block._get_fields_to_save())  # pylint: disable=protected-access
+
+        # We've pre-loaded the fields for this block, so the FieldData shouldn't
+        # consider these values "changed" in its sense of "you have to persist
+        # these because we've altered the field values from what was stored".
+        self.system.authored_data_store.mark_unchanged(block)
+
+        return block
+
+    def save_block(self, block):
+        """
+        Save any pending field data values to Learning Core data models.
+
+        This gets called by block.save() - do not call this directly.
+        """
+        if not self.system.authored_data_store.has_changes(block):
+            return  # No changes, so no action needed.
+
+        # Verify that the user has permission to write to authored data in this
+        # learning context:
+        if self.user is not None:
+            learning_context = get_learning_context_impl(block.scope_ids.usage_id)
```

Comment:
@ormsbee This returns a subclass of LearningContext, not an instance. Perhaps you just want `learning_context = block.scope_ids.usage_id.context_key`?

Reply:
I lifted this straight out of the blockstore implementation:

https://github.com/openedx/edx-platform/blob/5a36fa9163dd2608d4089a45ff64c0dec6bc0c9a/openedx/core/djangoapps/xblock/runtime/blockstore_runtime.py#L131-L135

But `block.scope_ids.usage_id.context_key` is the `LearningContextKey` right? Where I need to ultimately get a [LibraryContextImpl](https://github.com/openedx/edx-platform/blob/5a36fa9163dd2608d4089a45ff64c0dec6bc0c9a/openedx/core/djangoapps/content_libraries/library_context.py#L21)?